docs/src/main/asciidoc/centralized-log-management.adoc

```diff
@@ -236,6 +236,83 @@ networks:
 
 Launch your application, you should see your logs arriving inside the Elastic Stack; you can use Kibana available at http://localhost:5601/ to access them.
 
+
+[[logstash_ecs]]
+== GELF alternative: Send logs to Logstash in the ECS (Elastic Common Schema) format
+
+You can also send your logs to Logstash using a TCP input in the https://www.elastic.co/guide/en/ecs-logging/overview/current/intro.html[ECS] format.
+To achieve this we will use the `quarkus-logging-json` extension to format the logs in JSON format and the socket handler to send them to Logstash.
+
+For this you can use the same `docker-compose.yml` file as above but with a different Logstash pipeline configuration.
+
+[source]
+----
+input {
+  tcp {
+    port => 4560
+    coded => json
+  }
+}
+
+filter {
+  if ![span][id] and [mdc][spanId] {
+    mutate { rename => { "[mdc][spanId]" => "[span][id]" } }
+  }
+  if ![trace][id] and [mdc][traceId] {
+    mutate { rename => {"[mdc][traceId]" => "[trace][id]"} }
+  }
+}
+
+output {
+  stdout {}
+  elasticsearch {
+    hosts => ["http://elasticsearch:9200"]
+  }
+}
+----
+
+Then configure your application to log in JSON format instead of GELF
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-logging-json</artifactId>
+</dependency>
+----
+
+and specify the host and port of your Logstash endpoint. To be ECS compliant, some keys need to be renamed.
+
+[source, properties]
+----
+# to keep the logs in the usual format in the console
+quarkus.log.console.json=false
+
+quarkus.log.socket.enable=true
+quarkus.log.socket.json=true
+quarkus.log.socket.endpoint=localhost:4560
+
+# to have the exception serialized into a single text element
+quarkus.log.socket.json.exception-output-type=formatted
+
+# override some keys to be ECS compliant
+quarkus.log.socket.json.key-overrides=timestamp=@timestamp,\
+	logger_name=log.logger,\
+	level=log.level,\
+	process_id=process.pid,\
+	process_name=process.name,\
+	thread_name=process.thread.name,\
+	thread_id=process.thread.id,\
+	sequence=event.sequence,\
+	host_name=host.hostname,\
+	stack_trace=error.stack_trace
+quarkus.log.socket.json.excluded-keys=loggerClassName
+quarkus.log.socket.json.additional-field."service.environment".value=${quarkus.profile}
+quarkus.log.socket.json.additional-field."service.name".value=${quarkus.application.name}
+quarkus.log.socket.json.additional-field."service.version".value=${quarkus.application.version}
+quarkus.log.socket.json.additional-field."data_stream.type".value=logs
+quarkus.log.socket.json.additional-field."ecs.version".value=1.12.2
```

Comment:
This I just don't like. Could we at least have a enum property on the JSON formatter for format presets? Asking every user to provide all of this each time is not OK IMO.

Reply:
that's indeed what I proposed here: https://github.com/quarkusio/quarkus/pull/43232#issuecomment-2352186654